### ISSUE TITLE
Debian 11 Support

### ISSUE THREAD

**Issue by cforml** [datto/dattobd#277]
Any update on this? Here it is May 2022 and there is still no support for Debian 11. What is taking so long to get this working? Its been almost a year since Debian 11 has been released.

**Comment by kslen** [comment]
Would it maybe be possible to get a crowdfunding effort going for this?

**Comment by dakotarwilliams** [comment]
We already have some significant work toward kernel 5.9 compatibility. The issue is that 5.9 turned off many of the interfaces used for single-queue i/o that was dominant in the kernel at the time that dattobd was originally developed. There's a lot of work involved in properly adapting the project to the blk-mq model which is what the Linux kernel community has moved toward. We understand that modern kernel compatibility is important, especially considering that there are many "stable" distros producing new releases that we'd want to support. With this and renewed interest in the project internally, kernel compatibility is our number 1 priority. It's just a lot of work.

**Comment by nickchen-cpu** [comment]
Agreed, there are some kernel code check if `block_device->gen_disk->fops->submit_bio` exists to route `mq` logic.

**Comment by cforml** [comment]
@dakotarwilliams ok thank you for the update. I will wait to hear more updates on this.

**Comment by Fantu** [comment]
hi, I found recently [blksnap](https://github.com/veeam/blksnap) that have a kernel module for block-level snapshot for backup that don't use internal functions linked for address and don't have this issue
they are also trying to integrating upstream:
https://lore.kernel.org/linux-block/1655135593-1900-1-git-send-email-sergei.shtepa@veeam.com/
https://github.com/veeam/blksnap/issues/2
@SergeiShtepa and developers of this project: I think that that it would be useful to combine forces for a unique and best solution, at least upstream for the kernel module, or even better if you will also want library and tools
I already saw the good result recently for example in https://github.com/memtest86plus/memtest86plus where the union of memtest86+ and [pcmemtest](https://github.com/martinwhitaker/pcmemtest) projects and developers led to a better result
